Field-Robotics-Lab/glider_hybrid_whoi
# Dead reckoning standalone node.

Adds a standalone ROS node to emulate the dead reckoning of the for a glider.  Implements the algorithm and addresses https://github.com/Field-Robotics-Lab/glider_hybrid_whoi/issues/3

### Theory of Operation

* Subscribes to GPS, IMU and pressure sensor publications
* Publishes a lat/lon estimate using a NavSatFix message
* When there is a GPS fix, just reports that fix (no filtering)
* When there is no GPS fix, uses the dead reckoning algo. from https://github.com/Field-Robotics-Lab/glider_hybrid_whoi/issues/3

Initialization and usage parameters are illustrated in the `deadreckon.launch` file.

### How to test/demo

1. Start simulation:  `roslaunch glider_hybrid_whoi_gazebo start_demo_kinematics.launch`
2. Start the node: `roslaunch glider_deadreckoning deadreckon.launch`

The current simulation example doesn't seem to have a pressure sensor, so the dead reckoning isn't exercises (the velocities are all zero.  Would need to have a pressure/depth sensor input to verify the algorithm.
